interfaces/builtin/fwupd.go

```diff
@@ -64,13 +68,78 @@ const fwupdPermanentSlotAppArmor = `
   /sys/firmware/efi/esrt/entries/** r,
 
   # Allow fwupd to access system information
-  /sys/devices/virtual/dmi/id/product_name r,
-  /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/smbios_entry_point r,
+  /sys/devices/virtual/dmi/id/{sys,bios,product,chassis,board}_{name,version,vendor,date,serial,sku,type,asset_tag,family} r,
+
+  /sys/firmware/dmi/tables/DMI r,
+  /sys/bus/usb/devices/ r,
+
+  /sys/firmware/acpi/bgrt/{type,version,{x,y}offset,image} r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
+  /proc/modules r,
+  /run/udev/data/* r,
+  /sys/devices/**/usb*/busnum r,
+  /sys/devices/**/usb*/**/busnum r,
+  /sys/devices/**/usb*/devnum r,
+  /sys/devices/**/usb*/**/devnum r,
+  /sys/devices/**/usb*/speed r,
+  /sys/devices/**/usb*/**/speed r,
+  /sys/devices/**/usb*/descriptors r,
+  /sys/devices/**/usb*/**/descriptors r,
+  /sys/devices/**/usb*/bConfigurationValue r,
+  /sys/devices/**/usb*/**/bConfigurationValue r,
+  /dev/mem r,
+  /sys/kernel/security/lockdown r,
+  owner @{PROC}/@{pid}/mountinfo r,
+  /sys/bus/pci/devices/ r,
+  /sys/devices/pci*/**/vendor r,
+  /sys/devices/pci*/**/device r,
+  /sys/devices/pci*/**/revision r,
+  /sys/devices/pci*/**/subsystem_vendor r,
+  /sys/devices/pci*/**/subsystem_device r,
+  /sys/devices/pci*/**/class r,
+  /sys/devices/pci*/**/config r,
+  /sys/devices/pci*/**/name r,
+  /sys/devices/**/serio*/description r,
+  /dev/cpu/*/msr rw,
+
```

Comment:
Some more things you'll need that I know about.
for amd_pmc plugin:
```
/sys/devices/platform/**/smu_fw_version
/sys/devices/platform/**/smu_program
```
for pci-psp plugin:
```
/sys/devices/pci*/**/anti_rollback_status
/sys/devices/pci*/**/fused_part
/sys/devices/pci*/**/hsp_tpm_available
/sys/devices/pci*/**/rom_armor_enforced
/sys/devices/pci*/**/rpmc_production_enabled
/sys/devices/pci*/**/rpmc_spirom_available
/sys/devices/pci*/**/tsme_status
```

These two don't yet exist but are planning to come later.  To avoid a second spin maybe you want to add them now?  Up to you.
```
/sys/devices/pci*/**/bootloader_version
/sys/devices/pci*/**/tee_version
```

Reply:
I suppose those only need read.

Reply:
> I suppose those only need read.

Yes

Reply:
Added